natchez-datadog/src/main/scala/com/ovoenergy/effect/natchez/SubmittableSpan.scala

```diff
@@ -0,0 +1,138 @@
+package com.ovoenergy.effect.natchez
+
+import java.util.concurrent.TimeUnit.NANOSECONDS
+
+import cats.Applicative
+import cats.effect.{Clock, ExitCase}
+import cats.syntax.apply._
+import com.ovoenergy.effect.natchez.DatadogTags.{SpanType, forThrowable}
+import io.circe.{Decoder, Encoder}
+import io.circe.Encoder.encodeString
+import io.circe.generic.extras.Configuration
+import io.circe.generic.extras.semiauto._
+import natchez.TraceValue
+import natchez.TraceValue.StringValue
+
+/**
+ * This is the type we need to send to the Datadog agent
+ * to submit traces. Turning a span into one of these involves some trickery
+ * with tags, hence this being its own file now
+ */
+case class SubmittableSpan(
+  traceId: Long,
+  spanId: Long,
+  name: String,
+  service: String,
+  resource: String,
+  `type`: Option[SpanType],
```

Comment:
how come this is an option?

Reply:
Any spans we don't explicitly add a type to become `Custom` - which is also why I don't turn the `Custom` span type into a string when submitting spans to Datadog